* Nama : **Muhammad Farras Ma'ruf**
* NIM : **241012050065**
* Hal : **Ujian AKhir Semester**
* Mata Kuliah : **Statistik dan Data Analisis**
* Dosen Pengampu : **Dr Tukiyat**

**Dokumen Pendukung**
Berikut juga saya lampirkan dokumen pendukung seperti source code dan data yang saya gunakan dalam membuat analisa atas soal UAS

* [File source code ipynb](#)
* Data

## SOAL 1

![image.png](./assets/soal1.png)

### Menyiapkan Data

In [35]:
data_soal1 = pd.read_csv('./data/data_1.csv')
data_soal1 = data_soal1.drop(columns=['Unnamed: 0'])
data_soal1.head()# Print 5 data awal

nama_mahasiswa  skor  selisih
0         Farras     4        1
1          Faris     5        2
2           Noah     3        0
3          Tania     4        1
4         Hilyah     4        1

In [42]:
data_soal1.describe().head()

skor    selisih
count  20.000000  20.000000
mean    3.950000   0.950000
std     0.944513   0.944513
min     2.000000  -1.000000
25%     3.750000   0.750000

DataFrame yang digunakan pada uji **Wilxocon One sample** terdiri dari 20 baris data yang terdiri dari informasi Nama Mahasiswa, Tingkat Kepusan mahasiswa dan selisih standard nilai kepuasan dengan skor tingkat kepuasan dengan rentang nilai 1-5 dimana 1 tidak puas hingga 5 sangat puas.

## Jawaban Soal 1 : A

$H_0$ (Hipotesis Nol): $Median = 3$ (Tidak ada perbedaan yang signifikan antara median tingkat kepuasan mahasiswa dengan nilai standar 3).

$H_1$ (Hipotesis Alternatif): $Median \neq 3$ (Terdapat perbedaan yang signifikan antara median tingkat kepuasan mahasiswa dengan nilai standar 3).

## Jawaban Soal 1 : B (Uji statistik)

In [40]:
stat_uji, p_value = stats.wilcoxon(data_soal1.selisih, alternative='two-sided')

# 5. Menampilkan Statistik Deskriptif Sederhana
median_aktual = np.median(data_sampel)

## Jawaban Soal 1 : C dan D (Hasil uji dan interpretasi)

In [41]:
print(f"Median Sampel: {median_aktual}")
print(f"Nilai Standar: {nilai_standar}")
print("-" * 30)
print(f"Nilai Statistik Uji (W): {stat_uji}")
print(f"Nilai P-Value: {p_value:.5f}")

Median Sampel: 4.0
Nilai Standar: 3
------------------------------
Nilai Statistik Uji (W): 12.0
Nilai P-Value: 0.00153


Menurut perhitungan yang saya lakukan diatas, maka;

1. Nilai Statistik Uji (W):Nilai ini merupakan jumlah terkecil dari ranking bertanda (signed rank). Dalam contoh simulasi di atas, nilai W kemungkinan kecil karena data condong positif (jauh dari netral).Contoh Hasil: **12.0**
2. Nilai P-Value:Probabilitas memperoleh hasil ekstrem jika $H_0$ benar.Hasil: **0.000153** (Sangat kecil)
3. Keputusan Uji:Kita membandingkan P-value dengan tingkat signifikansi ($\alpha$), biasanya 0.05 (5%).Aturan Keputusan:
* Jika P-value < 0.05 $\rightarrow$ Tolak $H_0$.
* Jika P-value > 0.05 $\rightarrow$ Gagal Tolak $H_0$.

Dalam simulasi di atas: Karena $0.00043 < 0.05$, maka Tolak $H_0$.

Dari hasil uji statistik Wilxocon One Sample diatas melihat apakah statndar kepuasan = 3 atau tidak, akan tetapi kita tidak mengetahui araah-nya. Karena nilai tersebut tidak sama dnegan 3, dengan membandingkan dengan nilai media sample dengan nilai **4** yang mana lebih tinggai dari nilai 3 maka dapat disimpulkan _**kinerja layanan akademik perguruan tinggi tersebut dinilai baik dan memuaskan oleh mahasiswa secara signifikan**_.

## Soal 2

![Soal 2](./assets/soal2.png)

### Menyiapkan Data

In [38]:
# Memuat data
df_gabungan = pd.read_csv('./data/data_2.csv')
df_gabungan = df_gabungan.drop(columns=['Unnamed: 0'])
df_gabungan.head()

Nama     Gender  Skor Kepuasan
0    Budi  Laki-laki              3
1     Eko  Laki-laki              4
2   Fajar  Laki-laki              2
3  Hendra  Laki-laki              4
4    Joko  Laki-laki              3

In [43]:
df_gabungan.describe().head()

Skor Kepuasan
count      24.000000
mean        3.750000
std         0.989071
min         2.000000
25%         3.000000

DataFrame yang digunakan pada uji **Mann-Whitney U (Dua Kelompok Independen)** terdiri dari 4 baris data yang terdiri dari informasi Nama Mahasiswa, jenis kelamin dan nilai kepuasan dengan skor tingkat kepuasan dengan rentang nilai 1-5 dimana 1 tidak puas hingga 5 sangat puas.

## Jawaban Soal 2 : A

Variable dependen disini adalah **Skor kepusan** sedangkan variable independen adalah jenis **kelamin mahasiswa**. Dengan detail jawaban saya sebagai berikut;abs

**Variabel Independen**
* Adalah: Jenis Kelamin (Gender)
* Kategori: Laki-laki dan Perempuan.

**Variabel Dependen**
* Adalah: Tingkat Kepuasan Mahasiswa
* Data: Skor Skala Likert (1 - 5).

## Jawaban Soal 2 : B (Uji Mann-Whitney)

In [46]:
stat_u, p_value = stats.mannwhitneyu(data_laki, data_perempuan, alternative='two-sided')

In [44]:
# Menghitung Z-score secara manual dari nilai U
n1 = len(data_laki)
n2 = len(data_perempuan)

# Mean ekspektasi U
mu_u = (n1 * n2) / 2

# Standar deviasi U
sigma_u = np.sqrt((n1 * n2 * (n1 + n2 + 1)) / 12)

# Menghitung Z
z_score = (stat_u - mu_u) / sigma_u

## Jawaban Soal 2 : C (Menentukan nilai Z dan p-value)

In [47]:
print("\n--- Hasil Uji Mann-Whitney ---")
print(f"Statistik U: {stat_u}")
print(f"P-Value    : {p_value:.5f}")

print("\n--- Perhitungan Detail ---")
print(f"Nilai Z      : {z_score:.4f}")
print(f"Nilai P-Value: {p_value:.5f}")


--- Hasil Uji Mann-Whitney ---
Statistik U: 31.5
P-Value    : 0.01580

--- Perhitungan Detail ---
Nilai Z      : -2.3383
Nilai P-Value: 0.01580


## Jawaban Soal 2 : D (Interpretasi)

Dari hasi P-Value uji statistik diatas kita mendapatkan nilai **0.01580** dan bandingkan dengan Alpha (0.05).

Dengan hipotesa yang telah saya definisikan diatas;
* $H_0$: Tidak terdapat perbedaan signifikan pada tingkat kepuasan antara mahasiswa laki-laki dan
perempuan.
* $H_1$: Terdapat perbedaan signifikan pada tingkat kepuasan antara mahasiswa laki-laki dan perempuan.

Karena P-value **0.01580** < 0.05 :maka: Tolak $H_0$.Interpretasi: Terdapat perbedaan tingkat kepuasan yang signifikan antara mahasiswa laki-laki dan perempuan. Jenis gender tidak memliki pengaruh tingkat kepuasan karena bisa jadi apa yang diberikan oleh kampus dari segi fasilitas dan pendidikan pada mahasiswa laki=laki atau perempuan sama rata.

## Soal 3

![Soal 3](./assets/soal3.png)

### Menyiapkan data

In [61]:
df_pembelajaran = pd.read_csv('./data./daa_3.csv')
df_pembelajaran = df_pembelajaran.drop(columns=['Unnamed: 0'])
df_pembelajaran.head()

Metode  Skor Kepuasan
0  Tatap Muka              5
1  Tatap Muka              4
2  Tatap Muka              5
3  Tatap Muka              4
4  Tatap Muka              5

## Jawaban Soal 3 : A (Alasan Penggunaan Uji Kruskal–Wallis)
Uji Kruskal-Wallis digunakan karena alasan berikut:

1. Tipe Data Ordinal: Data kepuasan diukur menggunakan Skala Likert (1-5), yang merupakan data ordinal (berjenjang), bukan interval/rasio murni. Oleh karena itu, uji parametrik seperti ANOVA (Analysis of Variance) kurang tepat.

2. Lebih dari Dua Kelompok: Kita membandingkan 3 kelompok independen (Tatap Muka, Daring, Hybrid). Jika hanya 2 kelompok, kita menggunakan Mann-Whitney; jika lebih dari 2, kita gunakan Kruskal-Wallis.

3. Distribusi Data: Uji ini adalah uji non-parametrik yang tidak mewajibkan data berdistribusi normal (bebas distribusi).

## Jawaban Soal 3 : B (Rumusan Hipotesa)

* $H_0$ (Hipotesis Nol): Median tingkat kepuasan mahasiswa sama pada ketiga metode pembelajaran (Tidak ada perbedaan signifikan antar metode).
* $H_1$ (Hipotesis Alternatif): Minimal ada satu metode pembelajaran yang memiliki median tingkat kepuasan yang berbeda (Terdapat perbedaan signifikan antar metode).

## Jawaban Soal 3 : C & D (Pengujian dan Kesimpulan)

In [62]:
# Menyiapkan data per kelompok untuk fungsi kruskal
grup_tm = df_pembelajaran[df_pembelajaran['Metode'] == 'Tatap Muka']['Skor Kepuasan']
grup_daring = df_pembelajaran[df_pembelajaran['Metode'] == 'Daring']['Skor Kepuasan']
grup_hybrid = df_pembelajaran[df_pembelajaran['Metode'] == 'Hybrid']['Skor Kepuasan']

# Melakukan Uji Kruskal-Wallis
stat_h, p_value = stats.kruskal(grup_tm, grup_daring, grup_hybrid)

print(f"Nilai H-Statistic: {stat_h:.4f}")
print(f"Nilai P-Value    : {p_value:.5f}")

# Logika Pengambilan Keputusan Otomatis
alpha = 0.05
print("\n--- KESIMPULAN ---")
if p_value < alpha:
    print("Keputusan: Tolak H0")
    print("Interpretasi: Terdapat perbedaan tingkat kepuasan yang SIGNIFIKAN antara metode Tatap Muka, Daring, dan Hybrid.")
else:
    print("Keputusan: Gagal Tolak H0")
    print("Interpretasi: TIDAK terdapat perbedaan yang signifikan (tingkat kepuasan dianggap sama).")

Nilai H-Statistic: 12.8861
Nilai P-Value    : 0.00159

--- KESIMPULAN ---
Keputusan: Tolak H0
Interpretasi: Terdapat perbedaan tingkat kepuasan yang SIGNIFIKAN antara metode Tatap Muka, Daring, dan Hybrid.


Nilai P-Value: **0.00159**, jauh di bawah 0.05. Sehingga dapat saya simpulkan: Karena $P-value < 0.05$, maka Menolak $H_0$.Implikasi: Secara statistik terbukti bahwa metode pembelajaran mempengaruhi kepuasan mahasiswa.